File: 2015/AI/AIClient_JAVA/src/event/UpdateCharacterEvent.java
/* ------------------------------------------------------------------------------
 ** _________   _________      ________    _____      _____  ___________ _________
 ** \_   ___ \ /   _____/     /  _____/   /  _  \    /     \ \_   _____//   _____/
 ** /    \  \/ \_____  \     /   \  ___  /  /_\  \  /  \ /  \ |    __)_ \_____  \ 
 ** \     \____/        \    \    \_\  \/    |    \/    Y    \|        \/        \
 **  \______  /_______  /     \______  /\____|__  /\____|__  /_______  /_______  /
 **         \/        \/             \/         \/         \/        \/        \/ 
 **
 ** UpdateCharacterEvent.java
 ** Event that happen when a character move
 **
 ** Author: Samuel-Ricardo Carriere
 ** ------------------------------------------------------------------------------*/

package event;

import mathUtils.Vector2;
import world.World;

public class UpdateCharacterEvent extends IngoingEvent {
	private int teamID, characterID;
	private int positionX, positionY;

	@Override
	public void execute() {
		//System.out.println("Move character:" + characterID + " from " + teamID
		//		+ " to " + positionX + "," + positionY);
		Vector2 position = new Vector2(positionX, positionY);
		World.getInstance().getTeam(teamID).getCharacter(characterID)
				.updateInfo(position);
	}

	@Override
	public boolean fillArguments(String message) {
		String[] messageParts = message.split(SEPARATOR, 5);

		teamID = convertCharToNumeral(messageParts[0]);
		if (teamID == 0) {
			return false;
		}

		characterID = convertCharToNumeral(messageParts[1]);
		positionX = convertCharToNumeral(messageParts[2]);
		positionY = convertCharToNumeral(messageParts[3]);
		return true;
	}

}

File: 2015/AI/AIClient_Python/src/event/UpdateCharacterEvent.py
'''
Created on Dec 21, 2014

@author: scarriere
'''
from event.IngoingEvent import IngoingEvent
from world.World import World
from aiclient.Singleton import Singleton
from mathUtils.Vector2 import Vector2

class UpdateCharacterEvent(IngoingEvent):

    def __init__(self):
        self.teamId = 0
        self.characterId = 0
        self.position = Vector2(0,0)
    
    def fillArguments(self, string):
        arguments = string.split(":", 4)
        
        self.teamId = int(arguments[0])
        self.characterId = int(arguments[1])
        self.position.x = int(arguments[2])
        self.position.y = int(arguments[3])
        
    def execute(self):
        world = Singleton(World)
        world.getTeam(self.teamId).characters[self.characterId]._updatePosition(self.position)
        